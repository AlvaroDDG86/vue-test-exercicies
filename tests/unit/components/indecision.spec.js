import { shallowMount } from '@vue/test-utils';
import Indecision from '@/components/Indecision'
describe('Indecision', () => {
    let wrapper
    let getAnswerSpy

    // Mock the fetch api, doesn't exist in node
    global.fetch = jest.fn()

    beforeEach(() => {
        wrapper = shallowMount(Indecision)
        getAnswerSpy = jest.spyOn(wrapper.vm, 'getAnswer')
        jest.clearAllMocks()
    })

    // Tests
    test('should match with snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    });
    
    test('should not fire fetch when typing', async () => {
        const inputModel = wrapper.find('[data-testid="input"')
        await inputModel.setValue('Hello world')
        expect(getAnswerSpy).not.toHaveBeenCalled()
    });

    test('should fire the fetch when type the "?"', async () => {
        const inputModel = wrapper.find('[data-testid="input"')
        await inputModel.setValue('Hello world?')
        expect(getAnswerSpy).toHaveBeenCalled()
    });
    
    test('should run getAnswer', () => {
        // TODO
    });
    
    test('should run error on getAnswer', () => {
        // TODO
    });
});