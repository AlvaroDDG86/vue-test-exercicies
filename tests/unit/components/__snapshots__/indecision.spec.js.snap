// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indecision should match with snapshot 1`] = `
<!--v-if-->
<div class="bg-dark"></div>
<div class="indecision-container"><input data-testid="input" type="text" placeholder="Hazme una pregunta">
  <p>Recuerda terminar con un signo de interrogaci√≥n (?)</p>
  <!--v-if-->
</div>
`;
